scripts/test.sh

```diff
@@ -0,0 +1,97 @@
+#!/bin/bash
+
+# Set current project root
+PROJ_ROOT=$(pwd)
+
+# Get command line arguments
+while getopts :uisl OPTION
+do
+    case "${OPTION}" in
+        u) RUN_OSS_UNIT=true;;
+        i) RUN_OSS_INTEG=true;;
+        s) RUN_OSS_STYLE=true;;
+        l) RUN_LOCAL=true;;
+        ?) printf "script usage: $(basename $0) [-u] [-i] [-s] [-l]\n"; exit 1;;
+    esac
+done
+
+if [ "$RUN_LOCAL" = true ]; then
+    CYPRESS_ENV_VAR='true';
+else
+    CYPRESS_ENV_VAR='false';
+fi
+
+TEST_NAMES=()
+TEST_RESULTS=()
+NUM_TESTS_PASSED=0
+
+updateNumTestsPassed() {
+    if [ "$1" = true ]; then
+        NUM_TESTS_PASSED=$(($NUM_TESTS_PASSED+1))
+    fi
+}
+
+# Run unit tests
+if [ "$RUN_OSS_UNIT" = true ]; then
+    printf "\n======== Running OSS unit tests ========\n"
+    yarn --cwd "${PROJ_ROOT}/vscode" test-browser --browser chromium
+    PASS_UNIT_TESTS=$([[ $? -eq 0 ]] && echo true || echo false)
+
+    TEST_NAMES+=("OSS Unit Tests")
+    TEST_RESULTS+=($PASS_UNIT_TESTS)
+    updateNumTestsPassed $PASS_UNIT_TESTS
+fi
+
+# Run integration tests
+if [ "$RUN_OSS_INTEG" = true ]; then
+    printf "\n======== Running OSS integration tests ========\n"
+    yarn --cwd "${PROJ_ROOT}/vscode/test/integration/browser"
+    yarn --cwd "${PROJ_ROOT}/vscode/test/integration/browser" compile
+    sh ${PROJ_ROOT}/vscode/scripts/test-web-integration.sh --browser chromium
+    PASS_INTEG_TESTS=$([[ $? -eq 0 ]] && echo true || echo false)
+
+    TEST_NAMES+=("OSS Integration Tests")
+    TEST_RESULTS+=($PASS_INTEG_TESTS)
+    updateNumTestsPassed $PASS_INTEG_TESTS
+fi
+
+# Run style checks
+if [ "$RUN_OSS_STYLE" = true ]; then
+    printf "\n======== Running OSS style check ========\n"
+    printf "\nNOT IMPLEMENTED\n"
+    # PASS_STYLE_CHECK=$([[ $? -eq 0 ]] && echo true || echo false)
+    PASS_STYLE_CHECK=false
+
+    TEST_NAMES+=("OSS Style Check")
+    TEST_RESULTS+=($PASS_STYLE_CHECK)
+    updateNumTestsPassed $PASS_STYLE_CHECK
```

Comment:
Should we comment this out? Looks like this would always cause the tests to fail until we implement style checks.

Reply:
this has been updated to run the actual style checks. user can choose to run the style checks. as of now, the codebase will fail those checks. but the style check works.